src/components/dialogs/create-case-dialog/create-case-dialog.tsx

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useIntl } from 'react-intl';
+import React, { useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { keyGenerator } from '../../../utils/functions';
+import { createCase } from '../../../utils/rest-api';
+import { HTTP_UNPROCESSABLE_ENTITY_STATUS } from '../../../utils/UIconstants';
+import { Grid } from '@mui/material';
+import {
+    addUploadingElement,
+    removeUploadingElement,
+} from '../../../redux/actions';
+import UploadNewCase from '../commons/upload-new-case';
+import { ElementType } from '../../../utils/elementType';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { NameCheckReturn, useNameCheck } from '../commons/use-name-check';
+import { useForm } from 'react-hook-form';
+import { CASE_FILE, CASE_NAME, DESCRIPTION } from '../../utils/field-constants';
+import { ErrorInput, TextInput, FieldErrorAlert } from '@gridsuite/commons-ui';
+import { yupResolver } from '@hookform/resolvers/yup/dist/yup';
+import CustomMuiDialog from '../custom-mui-dialog';
+import {
+    createCaseDialogFormValidationSchema,
+    getCreateCaseDialogFormValidationDefaultValues,
+} from './create-case-dialog-utils';
+
+const MAX_FILE_SIZE_IN_MO = 100;
+const MAX_FILE_SIZE_IN_BYTES = MAX_FILE_SIZE_IN_MO * 1024 * 1024;
+
+interface ICreateCaseDialogProps {
+    onClose: () => void;
+    open: boolean;
+}
+
+const CreateCaseDialog: React.FunctionComponent<ICreateCaseDialogProps> = ({
+    onClose,
+    open,
+}) => {
+    const intl = useIntl();
+    const dispatch = useDispatch();
+    const { snackError } = useSnackMessage();
+
+    const createCaseFormMethods = useForm<{
+        [CASE_NAME]: string;
+        [DESCRIPTION]: string;
+        [CASE_FILE]: File | null;
+    }>({
+        mode: 'onChange',
+        defaultValues: getCreateCaseDialogFormValidationDefaultValues(),
+        resolver: yupResolver(createCaseDialogFormValidationSchema),
+    });
+
+    const {
+        setValue,
+        formState: { errors },
+        setError,
+        getValues,
+        clearErrors,
+    } = createCaseFormMethods;
+
+    const activeDirectory = useSelector((state: any) => state.activeDirectory);
+    const userId = useSelector((state: any) => state.user.profile.sub);
+
+    const caseNameErrorMessage = errors.caseName?.message;
+    const caseFileErrorMessage = errors.caseFile?.message;
+    const { caseName, caseFile } = getValues();
```

Comment:
Yes, delete this, this is not how to use getValues.

Reply:
I delete it but according to documentation we can can use getValues this way..
https://react-hook-form.com/docs/useform/getvalues

`const values = getValues() // { test: "test-input", test1: "test1-input" }`

Reply:
getValues is meant to get form values at a specific time
it will not render anything, which mean you can get obsolete values here

here it is working because there is a bug with "use-name-check" which renders the whole dialog when changed
otherwise, this will bug